include/dlaf/eigensolver/gen_eigensolver/impl.h

```diff
@@ -40,15 +46,39 @@ template <Backend B, Device D, class T>
 void GenEigensolver<B, D, T>::call(comm::CommunicatorGrid& grid, blas::Uplo uplo, Matrix<T, D>& mat_a,
                                    Matrix<T, D>& mat_b, Matrix<BaseType<T>, D>& eigenvalues,
                                    Matrix<T, D>& eigenvectors, const Factorization factorization) {
+#ifdef DLAF_WITH_HDF5
+  static std::atomic<size_t> num_gen_eigensolver_calls = 0;
+  std::stringstream fname;
+  fname << "generalized-eigensolver-" << matrix::internal::TypeToString_v<T> << "-"
+        << std::to_string(num_gen_eigensolver_calls) << ".h5";
+  std::optional<matrix::internal::FileHDF5> file;
```

Comment:
I think it need to provide different names depending on `factorization`.

Reply:
I don't think it's needed because every call will update the same `static std::atomic`, but considering also your suggestion below it might be good to name them differently. I'd suggest to keep the same counter though, since it is easier to track synchronous calls (C/Fortran API).